gax-java/gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -318,14 +318,26 @@ boolean cancelIfStale() {
         switch (this.state) {
           case IDLE:
             if (!idleTimeout.isZero() && waitTime >= idleTimeout.toMillis()) {
-              myError = new WatchdogTimeoutException("Canceled due to idle connection", false);
+              myError =
+                  new WatchdogTimeoutException(
+                      "Canceled due to idle connection. waitTime="
+                          + waitTime
+                          + "ms idleTimeout="
+                          + idleTimeout.toMillis()
+                          + "ms",
+                      false);
             }
             break;
           case WAITING:
             if (!waitTimeout.isZero() && waitTime >= waitTimeout.toMillis()) {
               myError =
                   new WatchdogTimeoutException(
-                      "Canceled due to timeout waiting for next response", true);
+                      "Canceled due to timeout waiting for next response. waitTime="
+                          + waitTime
+                          + "ms waitTimeout="
+                          + waitTimeout.toMillis()
+                          + "ms",
```

Comment:
Similar as above:

"Cancel RPC after waiting for a server response for X ms (threshold: Y ms)

Reply:
Done.